tests/blueprints/blueprints-list.directive.spec.js

```diff
@@ -0,0 +1,692 @@
+
+    $scope.blueprints = getBlueprints();
+    $scope.serviceCatalogs = getServiceCatalogs();
+    $scope.tenants = getTenants();
+
+    var htmlTmp = '<blueprints-list blueprints="blueprints" service-catalogs="serviceCatalogs" tenants="tenants"/>';
+
+    compileHTML(htmlTmp, $scope);
+  });
+
+  describe('Blueprints List', function() {
+    it('should have correct number of rows', function () {
+      var rows = element.find('.list-view-pf > .list-group-item');
+      expect(rows.length).to.eq($scope.blueprints.length);
+    });
+
+    it('should have gotten catalog names from catalog ids', function () {
+      var catName0 = element.find('#catalogName_0');
+      var catName1 = element.find('#catalogName_1');
+      var catName2 = element.find('#catalogName_2');
+
+      expect(catName0.html()).to.eq("Amazon Operations");
+      expect(catName1.html()).to.eq("OpenStack Operations");
+      expect(catName2.html()).to.eq("Unassigned");
+    });
+
+    it('should display the correct Visibility icons', function () {
+      var publicIcon = element.find('.fa.fa-globe');
+      var privateIcon = element.find('.pficon.pficon-private');
+      var tenantIcon = element.find('.pficon.pficon-tenant');
+
+      expect(publicIcon.length).to.eq(1);
+      expect(privateIcon.length).to.eq(1);
+      expect(tenantIcon.length).to.eq(1);
+    });
+
+    it('should enable/disable delete blueprint button', function () {
+      var rows = element.find('.list-view-pf > .list-group-item');
+      expect(rows.length).to.eq($scope.blueprints.length);
+
+      var disabledDeleteButton = element.find('.primary-action[disabled]');
+      expect(disabledDeleteButton.length).to.eq(1);
+
+      var checkboxes = rows.find('input');
+      expect(checkboxes.length).to.eq(3);
+
+      checkboxes[0].click();
+      $scope.$digest();
+
+      disabledDeleteButton = element.find('.primary-action[disabled]');
+      expect(disabledDeleteButton.length).to.eq(0);
+    });
+
+    it('should enable/disable inline publish buttons', function () {
+
+      var rows = element.find('.list-view-pf > .list-group-item');
+      expect(rows.length).to.eq($scope.blueprints.length);
+
+      // When 0 Items on Canvas, Publish button is disabled
+      var secondRowNumItems = element.find('#numItems_1');
+      var secondRowNumItemsStr = angular.element(secondRowNumItems[0]).html();
+      secondRowNumItemsStr = secondRowNumItemsStr.substr(0,secondRowNumItemsStr.indexOf(' '));
+      expect(secondRowNumItemsStr).to.eq('0');
+
+      var disabledPublishButton = angular.element(rows[1]).find('.btn.btn-default.disabled');
+      expect(disabledPublishButton.length).to.eq(1);
+
+      // Third row has 3 Items on Canvas, so publish button should be enabled
+      disabledPublishButton = angular.element(rows[2]).find('.btn.btn-default.disabled');
+      expect(disabledPublishButton.length).to.eq(0);
+    });
+  });
+
+  function getBlueprints() {
+    return [
+      {
```

Comment:
I like the idea, but it looks like it would take a little more work to implement:

http://stackoverflow.com/questions/17370427/loading-a-mock-json-file-within-karmaangularjs-test